packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -94,16 +95,35 @@ class FlutterDevice {
     if (targetPlatform == TargetPlatform.web_javascript) {
       Artifact platformDillArtifact;
       List<String> extraFrontEndOptions;
+      final PackageConfig packageConfig = await loadPackageConfigWithLogging(
+        globals.fs.file(buildInfo.packagesPath),
+        logger: globals.logger,
+        throwOnError: false,
+      ) ?? PackageConfig.empty;
       if (buildInfo.nullSafetyMode == NullSafetyMode.unsound) {
         platformDillArtifact = Artifact.webPlatformKernelDill;
         extraFrontEndOptions = buildInfo.extraFrontEndOptions;
-      } else {
+      } else if (buildInfo.nullSafetyMode == NullSafetyMode.sound) {
         platformDillArtifact = Artifact.webPlatformSoundKernelDill;
-        extraFrontEndOptions = <String>[
-          ...?buildInfo?.extraFrontEndOptions,
-          if (!(buildInfo?.extraFrontEndOptions?.contains('--sound-null-safety') ?? false))
-            '--sound-null-safety'
-        ];
+        extraFrontEndOptions =  buildInfo.extraFrontEndOptions;
+      } else {
+        final LanguageVersion languageVersion = target != null
+          ? packageConfig?.packageOf(Uri.parse(target))?.languageVersion
+          : null;
+        final bool isNullSafe = languageVersion != null && languageVersion.major >= 2 && languageVersion.minor >= 11;
```

Comment:
A null language version is supposed to mean opt in, fwiw

Reply:
TIL!

Reply:
Thanks - updated to be null safe if the language version is null or is gte 2.11

Reply:
How are we going to handle the 2.12 flip here? That will be the sdk where its actually turned on

Reply:
I'll file an issue to adjust the constraints. This is only an approximation of the autodetect logic, so its going to have some holes.

Reply:
https://github.com/flutter/flutter/issues/69416